pkgs/by-name/de/depthai-core/generic.nix

```
@@ -0,0 +1,315 @@
+{
+  lib,
+  stdenv,
+  cmake,
+  fetchFromGitHub,
+  fetchzip,
+  fetchpatch,
+  opencv4,
+  python3Packages,
+  clang-tools,
+  pkg-config,
+  libusb1,
+  boost186,
+  libpng,
+  libarchive,
+  httplib,
+  openssl,
+  protobuf_21,
+  cproto,
+  eigen,
+  yaml-cpp,
+  jsoncpp,
+  spdlog,
+  argparse,
+  nlohmann_json,
+  libnop,
+  mcap,
+  websocketpp,
+  mp4v2,
+  neargye-semver,
+  backward-cpp,
+  pcl,
+  git,
+  catch2_3,
+  xtensor,
+  bzip2,
+  lz4,
+  xz,
+  apriltag,
+  xlink,
+  magic-enum,
+  nanorpc,
+  fp16,
+  curl,
+  cpr,
+  xtl,
+  fmt,
+  zlib,
+  depthai-data,
+  ws-protocol,
+  gcc-unwrapped,
+}:
+
+let
+  arch = builtins.elemAt (builtins.split "-" stdenv.hostPlatform.system) 0;
+
+  catch2_3WithSharedLibs = catch2_3.overrideAttrs (oldAttrs: {
+    cmakeFlags = (oldAttrs.cmakeFlags or [ ]) ++ [
+      "-DBUILD_SHARED_LIBS=ON"
+    ];
+  });
+
+  # Latest commits are not compatible, use older version
+  ws-protocolCompat = ws-protocol.overrideAttrs (old: {
+    src = fetchFromGitHub {
+      owner = "foxglove";
+      repo = "ws-protocol";
+      rev = "e8440dbc988efb42ad206fd3e2e46f4eb2e5f5c0";
+      hash = "sha256-jPFSN2edGE0ge8AWFvIeXfoXn9JcAcLOTfsBcJq+dUo=";
+    };
+
+    patches = [ ./001-Ws-protocol-add-missing-libs.patch ];
+  });
+
+  prefixVar = "\${" + "prefix}";
+  # Latest commits are not compatible, use older version
+  xtensorCompat = xtensor.overrideAttrs (old: {
```

`old` is seemingly not used

```suggestion
  xtensorCompat = xtensor.overrideAttrs {
```